src/koino.zig

```diff
@@ -1,4 +1,37 @@
+const std = @import("std");
+
 pub const parser = @import("parser.zig");
 pub const Options = @import("options.zig").Options;
 pub const nodes = @import("nodes.zig");
 pub const html = @import("html.zig");
+
+/// Performs work using internalAllocator, and writes the result to a Writer.
+fn markdownToHtmlInternal(writer: anytype, internalAllocator: std.mem.Allocator, markdown: []const u8, options: Options) !void {
+    var doc = try parse(internalAllocator, markdown, options);
+    defer doc.deinit();
+
+    try html.print(writer, internalAllocator, options, doc);
+}
+
+/// Parses Markdown into an AST.  Use `deinit()' on the returned document to free memory.
+pub fn parse(internalAllocator: std.mem.Allocator, markdown: []const u8, options: Options) !*nodes.AstNode {
+    var p = try parser.Parser.init(internalAllocator, options);
+    defer p.deinit();
+    try p.feed(markdown);
+    return try p.finish();
+}
+
+/// Performs work with an ArenaAllocator backed by the page allocator, and allocates the result HTML with resultAllocator.
+pub fn markdownToHtml(resultAllocator: std.mem.Allocator, markdown: []const u8, options: Options) ![]u8 {
```

Comment:
Note: `options` and `markdown` are flipped here compared to before, as the general convention in Zig is to put `options` after all other arguments.

Reply:
Nice, ta!